bootstrap.py

```diff
@@ -426,19 +426,28 @@ def fully_update_the_box():
 
 # curl https://satellite.example.com:9090/ssh/pubkey >> ~/.ssh/authorized_keys
 # sort -u ~/.ssh/authorized_keys
-def install_foreman_ssh_key():
+def install_foreman_ssh_key(remote_url):
     """
     Download and install the Satellite's SSH public key into the foreman user's
-    authorized keys file, so that remote execution becomes possible.
+    authorized keys file location, so that remote execution becomes possible.
+    By default is ~/.ssh/authorized_keys
     """
     userpw = pwd.getpwnam(options.remote_exec_user)
-    foreman_ssh_dir = os.sep.join([userpw.pw_dir, '.ssh'])
-    foreman_ssh_authfile = os.sep.join([foreman_ssh_dir, 'authorized_keys'])
+    if options.remote_exec_authpath:
+        if os.path.isdir(options.remote_exec_authpath):
+            foreman_ssh_dir = options.remote_exec_authpath
+            foreman_ssh_authfile = options.remote_exec_authpath + "/authorized_keys"
+        else:
+            print_error("Foreman's SSH keys not installed. Directory where authorized_keys file must be located is not found: %s" % options.remote_exec_authpath)
+            return
+    else:
+        foreman_ssh_dir = os.sep.join([userpw.pw_dir, '.ssh'])
```

Comment:
can we make the default value of `options.remote_exec_authpath` to be `os.path.join(userpw.pw_dir, '.ssh')` then we could skip this if/else block.

note that I changed `os.sep.join([userpw.pw_dir, '.ssh'])` to `os.path.join(userpw.pw_dir, '.ssh')` which I think is more correct here.

Reply:
Actually from the first comment options.remote_exec_authpath is the complete path to authorizedKey file not a dir.